bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/categorylist/FeaturedCateogoryListImplTest.java

```diff
@@ -14,55 +14,101 @@
+import com.adobe.cq.sightly.WCMBindings;
 import com.day.cq.wcm.api.Page;
+import io.wcm.testing.mock.aem.junit.AemContext;
+import io.wcm.testing.mock.aem.junit.AemContextCallback;
 
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 public class FeaturedCateogoryListImplTest {
 
-    FeaturedCateogoryListImpl slingModel;
-    private static final String TEST_PRODUCT_PAGE_URL = "/content/test-category-page";
-    private CategoryTree queryCategory;
-    List<CategoryInterface> categories = new ArrayList<CategoryInterface>();
+    private GraphqlClient graphqlClient;
+    private FeaturedCateogoryListImpl slingModel;
+    private Query rootQuery;
+    private List<CategoryInterface> categories = new ArrayList<CategoryInterface>();
+    private static final String TEST_CATEGORY_PAGE_URL = "/content/test-category-page";
+    private static final String TEST_IMAGE_URL = "/magento/category/img/500_F_4437974_DbE4NRiaoRtUeivMyfPoXZFNdCnYmjPq_1.jpg";
+    private static final int TEST_CATEGORY = 3;
+    private static final String TEST_CATEGORY_NAME = "Equipment";
+
+    @Rule
+    public final AemContext context = createContext("/context/jcr-content.json");
 
     @Before
     public void setup() throws Exception {
-        this.slingModel = new FeaturedCateogoryListImpl();
         String json = IOUtils.toString(this.getClass()
             .getResourceAsStream("/graphql/magento-graphql-singlecategory-result.json"), StandardCharsets.UTF_8);
-        Query rootQuery = QueryDeserializer.getGson().fromJson(json, Query.class);
-        CategoryTree category = rootQuery.getCategory();
-        category.setPath(TEST_PRODUCT_PAGE_URL + ".3.html");
-        this.queryCategory = category;
-        this.categories.add(category);
-        Whitebox.setInternalState(this.slingModel, "categories", categories);
+        rootQuery = QueryDeserializer.getGson().fromJson(json, Query.class);
         Page categoryPage = mock(Page.class);
         when(categoryPage.getLanguage(false)).thenReturn(Locale.US);
-        when(categoryPage.getPath()).thenReturn("/content/test-category-page");
-        Whitebox.setInternalState(this.slingModel, "categoryPage", categoryPage);
+        when(categoryPage.getPath()).thenReturn(TEST_CATEGORY_PAGE_URL);
+        Resource resource = Mockito.spy(context.resourceResolver().getResource("/content/pageA"));
+        graphqlClient = Mockito.mock(GraphqlClient.class);
+        Page page = mock(Page.class);
+        MockSlingHttpServletRequest request = context.request();
+        SlingBindings slingBindings = (SlingBindings) context.request().getAttribute(SlingBindings.class.getName());
+        slingBindings.put(WCMBindings.WCM_MODE, new SightlyWCMMode(request));
+        slingBindings.put("currentPage", page);
+        slingBindings.put("resource", resource);
+        Integer categoryId[] = { TEST_CATEGORY };
+        Map<String, Object> pageProperties = new HashMap<>();
```

Comment:
In the future, please consider splitting your methods if they get too long. The code would be more readable.